lib/protocol/protocol_test.go

```diff
@@ -813,3 +813,20 @@ func TestClusterConfigAfterClose(t *testing.T) {
 		t.Fatal("timed out before Cluster Config returned")
 	}
 }
+
+func TestDispatcherToCloseDeadlock(t *testing.T) {
```

Comment:
// Verify that we don't deadlock when calling Close() from within one of the model callbacks (ClusterConfig).